src/wp-includes/class-wp-theme.php

```diff
@@ -1993,16 +2002,31 @@ private function set_pattern_cache( array $patterns ) {
 			'version'  => $this->get( 'Version' ),
 			'patterns' => $patterns,
 		);
-		wp_cache_set( 'wp_theme_patterns_' . $this->stylesheet, $pattern_data, 'theme_files' );
+
+		/** This filter is documented in wp-includes/theme.php */
+		$cache_expiration = (int) apply_filters( 'wp_cache_themes_persistently', self::$cache_expiration, 'theme_block_patterns' );
```

Comment:
I think we should do a bit of validation after the filter. For example, it could return `0` or even negative values, which should not be used.

I think the simplest solution is to reuse the original value if the filter returns something invalid. As in:
```php
if ( $cache_expiration <= 0 ) {
	$cache_expiration = self::$cache_expiration;
}
```

Reply:
Agreed, I have also added a `doing_it_wrong warning` in case someone adds a below 1 or a non-int value